Update Android Gradle Plugin to `8.11.0` and Gradle Wrapper to `8.14.3` + update default API target to `35` in tests

FilipeMarch opened the pull request:
Already tested here and it's working fine:
On `buildozer.spec`:
```
android.api = 36
android.minapi = 24
p4a.source_dir = /home/kivyschool/python-for-android
```
```
[INFO]:    Will compile for the following archs: arm64-v8a
[INFO]:    Found Android API target in $ANDROIDAPI: 36
[INFO]:    Found NDK dir in $ANDROIDNDK: /home/kivyschool/.buildozer/android/platform/android-ndk-r25b
[INFO]:    Found NDK version 25b
[DEBUG]:   
[DEBUG]:   	Welcome to Gradle 8.14.3!
[DEBUG]:   	
[DEBUG]:   	Here are the highlights of this release:
[DEBUG]:   	 - Java 24 support
[DEBUG]:   	 - GraalVM Native Image toolchain selection
[DEBUG]:   	 - Enhancements to test reporting
[DEBUG]:   	 - Build Authoring improvements
[DEBUG]:   
[DEBUG]:        BUILD SUCCESSFUL in 12s
[DEBUG]:        34 actionable tasks: 33 executed, 1 up-to-date
[DEBUG]:   
```

## Summary

- `gradle-wrapper.properties`: `distributionUrl` now points to `gradle‑8.14.3‑all.zip`
- `build.tmpl.gradle`: 
  - classpath updated from `'com.android.tools.build:gradle:8.1.1'` to `'com.android.tools.build:gradle:8.11.0'`
  - Repository mirror updated from `jcenter()` to `mavenCentral()`.

**Motivation**

I tried to compile my application using `android.api = 36` but started getting these logs:
```
[INFO]:    Detected highest available build tools version to be 36.0.0
[DEBUG]:   -> running gradlew clean assembleDebug
[DEBUG]:       WARNING: We recommend using a newer Android Gradle plugin to use compileSdk = 36
[DEBUG]:       
[DEBUG]:       This Android Gradle plugin (8.1.1) was tested up to compileSdk = 34.
[DEBUG]:       
[DEBUG]:       You are strongly encouraged to update your project to use a newer
[DEBUG]:       Android Gradle plugin that has been tested with compileSdk = 36.
[DEBUG]:       
[DEBUG]:       If you are already using the latest version of the Android Gradle plugin,
[DEBUG]:       you may need to wait until a newer version with support for compileSdk = 36 is available.
[DEBUG]:       
[DEBUG]:       To suppress this warning, add/update
[DEBUG]:           android.suppressUnsupportedCompileSdk=36
[DEBUG]:       to this project's gradle.properties.
[DEBUG]:       
[DEBUG]:       > Task :processDebugMainManifest
[DEBUG]:       /home/kivyschool/test/.buildozer/android/platform/build-arm64-v8a/dists/kivy_reloader/src/main/AndroidManifest.xml:55:18-50 Warning:
[DEBUG]:               android:extractNativeLibs should not be specified in this source AndroidManifest.xml file. See https://d.android.com/guide/topics/manifest/application-element#extractNativeLibs for more information.
[DEBUG]:       The AGP Upgrade Assistant can remove the attribute from the AndroidManifest.xml file and update the build file accordingly. See https://d.android.com/studio/build/agp-upgrade-assistant for more information.
[DEBUG]:       
[DEBUG]:       > Task :compileDebugJavaWithJavac
[DEBUG]:       Note: Some input files use or override a deprecated API.
[DEBUG]:       Note: Recompile with -Xlint:deprecation for details.
[DEBUG]:       
[DEBUG]:       Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
```

After [some investigation](https://developer.android.com/build/releases/gradle-plugin#api-level-support):
<img width="1480" height="270" alt="image" src="https://github.com/user-attachments/assets/a9760018-c75e-4c28-b61f-b0b886e28290" />
<img width="1513" height="809" alt="image" src="https://github.com/user-attachments/assets/086d1d65-a8a1-4cf6-8814-a6a52a6b2cd3" />

* Play Console enforces `targetSdkVersion ≥ 35`; this requires AGP ≥ 8.6 and Gradle ≥ 8.13.
* Gradle 8.14.3 satisfies the toolchain floor for AGP 8.11.0.
* `jcenter()` is deprecated; switching to `mavenCentral()` removes a dead dependency source.

FilipeMarch committed:
🔧 build: upgrade Gradle wrapper to `8.14.3`

* `distributionUrl` now points to `gradle‑8.14.3‑all.zip`

FilipeMarch committed:
🔧 build: bump AGP to `8.11.0`, replace `jcenter()` with `mavenCentral()`

misl6 commented:
I tried to build with a slightly different version (been more conservative), a few days ago, and also seemed fine to me.
I see the automated tests are failing, did you already tried to reproduce it locally?

UPDATE: I see the issue might be the current default target SDK, I guess we should just increase it. 😄

FilipeMarch committed:
Update `android-api` to 35 in `testapps/on_device_unit_tests/setup.py`

FilipeMarch committed:
Update target android.api to 35 in buildozer.spec

FilipeMarch committed:
Update `ANDROID_API_LEVEL` to 35 in `android.mk`

misl6 commented:
Checked test results after latest changes, `APP_PLATFORM := $(NDK_API)` in Application.mk might be missing for web view and library 😬

FilipeMarch committed:
Set `APP_PLATFORM` in `Application.mk` to use `NDK_API`

FilipeMarch commented:
Hum... got 32 passed, 1 failed here; it seems something went wrong on `service_library`

FilipeMarch committed:
Add `Application.mk` with `APP_PLATFORM` for service_library bootstrap

Hopefully [this](https://github.com/kivy/python-for-android/actions/runs/16309986846/job/46063689527?pr=3172) will pass

Novfensec commented:
Latest stable NDK to build for `android.api = 36` is `r28c`. So The default minimum should be r28c for newer builds. Also using `ndk>=r28` will force 16KiB alignement as default. 

To override this behavior we can do below:
**This may not work as android may reject .so files**

`CMakeLists.txt`:
```txt
set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,max-page-size=4096")
```

`Android.mk`:
```mk
LOCAL_LDFLAGS += -Wl,-z,max-page-size=4096
```

Overall we need to first add 16KiB support.

T-Dynamos commented:
@Novfensec SDL won't build in latest NDK until #3164 is pushed.

Ref: https://github.com/kivy/python-for-android/pull/3136#issuecomment-2758849003

misl6 commented:
> Hum... got 32 passed, 1 failed here; it seems something went wrong on `service_library`

APP_ABI := $(ARCH) in Application.mk of service_library is missing :)

FilipeMarch committed:
Define `APP_ABI` variable in `Application.mk` for arch targeting

misl6 reviewed (approved):
LGTM. Thank you!
